### PR TITLE
[WIP] (GH-533) Implemented ability to change language dynamically

### PR DESCRIPTION
**Pull request by AdmiringWorm**
This pull request adds support for changing between languages dynamically during runtime.

This is still very much a WIP, and some features currently do not work.
Such as:
- Loading the correct language during startup
- Translation of 'This Computer'
- Grabbing all supported languages
- Validation

### CODE REVIEW

File: Source/ChocolateyGui/Views/SettingsView.xaml [hunk 1]
```diff
@@ -75,14 +75,15 @@
...
-                                    <models:ChocolateyGUISetting Title="{x:Static lang:Resources.SettingsView_ToggleDefaultTileViewRemoteTitle}" Description="{x:Static lang:Resources.SettingsView_ToggleDefaultTileViewRemoteDescription}" Enabled="{Binding Source={StaticResource BindingProxy}, Path=Data.DefaultToTileViewForRemoteSource}" />
-                                    <models:ChocolateyGUISetting Title="{x:Static lang:Resources.SettingsView_UseDelayedSearchTitle}" Description="{x:Static lang:Resources.SettingsView_ToggleUseDelayedSearchDescription}" Enabled="{Binding Source={StaticResource BindingProxy}, Path=Data.UseDelayedSearch}" />
+                                    <metro:SplitButton ItemsSource="{Binding Source={StaticResource BindingProxy}, Path=Data.AllLanguages}" SelectedItem="{Binding Source={StaticResource BindingProxy}, Path=Data.UseLanguage}" />
```
Comment (punker76) [inline Source/ChocolateyGui/Views/SettingsView.xaml]: If you put  the SplitButton outside the ItemsControl then you don't need the BindingProxy and didn't get this output

```
System.Windows.Data Error: 26 : ItemTemplate and ItemTemplateSelector are ignored for items already of the ItemsControl's container type; Type='SplitButton'
```

Solution

```xaml
<ScrollViewer Margin="5 0" Grid.Row="1" Width="300" HorizontalAlignment="Left" VerticalScrollBarVisibility="Auto">
    <StackPanel Orientation="Vertical">
        <metro:SplitButton ItemsSource="{Binding AllLanguages}" SelectedItem="{Binding UseLanguage, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}" />
        <ItemsControl ItemTemplate="{StaticResource ChocolateyGUISettingTemplate}">
            <ItemsControl.Items>
                <models:ChocolateyGUISetting Title="{lang:Localize SettingsView_ToggleShowConsoleOutputTitle}" Description="{lang:Localize SettingsView_ToggleShowConsoleOutputDescription}" Enabled="{Binding Source={StaticResource BindingProxy}, Path=Data.ShowConsoleOutput}" />
                <models:ChocolateyGUISetting Title="{lang:Localize SettingsView_ToggleDefaultTileViewLocalTitle}" Description="{lang:Localize SettingsView_ToggleDefaultTileViewLocalDescription}" Enabled="{Binding Source={StaticResource BindingProxy}, Path=Data.DefaultToTileViewForLocalSource}" />
                <models:ChocolateyGUISetting Title="{lang:Localize SettingsView_ToggleDefaultTileViewRemoteTitle}" Description="{lang:Localize SettingsView_ToggleDefaultTileViewRemoteDescription}" Enabled="{Binding Source={StaticResource BindingProxy}, Path=Data.DefaultToTileViewForRemoteSource}" />
                <models:ChocolateyGUISetting Title="{lang:Localize SettingsView_UseDelayedSearchTitle}" Description="{lang:Localize SettingsView_ToggleUseDelayedSearchDescription}" Enabled="{Binding Source={StaticResource BindingProxy}, Path=Data.UseDelayedSearch}" />
            </ItemsControl.Items>
        </ItemsControl>
    </StackPanel>
</ScrollViewer>
```

Comment (punker76) [inline Source/ChocolateyGui/Views/SettingsView.xaml]: Why using a SplitButton?

Comment (AdmiringWorm) [reply Source/ChocolateyGui/Views/SettingsView.xaml]: No special reason, just something that popped into mind since I couldn't find a `DropDownList`.

Comment (AdmiringWorm) [reply Source/ChocolateyGui/Views/SettingsView.xaml]: thanks, I didn't notice that error.

Comment (punker76) [reply Source/ChocolateyGui/Views/SettingsView.xaml]: The ComboBox has a DropDownList, or did you have another requirement?

Comment (AdmiringWorm) [reply Source/ChocolateyGui/Views/SettingsView.xaml]: Oh right, completely forgot about ComboBox. Been a while since I last worked in WPF.

**Comment by AdmiringWorm** [comment]
A status update on this PR.

Unfortunately due to other priorities, and me not using Windows anymore as my main OS, this PR is currently on hold for me.
While I do plan do eventually set up a VM to continue working on this PR, this is not something I see myself doing in the near future. So if anyone wishes to continue working on this, please feel free to do so.

**Comment by mwallner** [comment]
Hey, any progress on this?
@AdmiringWorm do you want "us" to take this over?

**Comment by AdmiringWorm** [comment]
> Hey, any progress on this?

No, unfortunately not.

> @AdmiringWorm do you want "us" to take this over?

Feel free, I don't see that I will be able to continue on this anytime in the near future.

**Comment by gep13** [comment]
@AdmiringWorm I am going to go ahead and close this PR, as I have replayed the changes in this PR:

https://github.com/chocolatey/ChocolateyGUI/pull/844

Resolving the merge conflicts in this PR was going to be too difficult, so I just started again 😄

**Pull request closed without merging**